### PR TITLE
Tools: Add topology2 support for IIR and FIR EQ tool and add coefficients .conf data files

### PR DESCRIPTION
**Pull request by singalsu**

### CODE REVIEW

**Comment by singalsu** [comment]
This continues/complements from #5306. Still draft, I'm adding topologies to test.

File: tools/topology/topology2/include/components/eqiir/passthrough.conf [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Pass-through, created with example_iir_eq.m 25-Mar-2022
+Object.Base.data."eqiir_passthough" {
```
Comment (singalsu) [inline tools/topology/topology2/include/components/eqiir/passthrough.conf:2]: Typo "passthough", fix this and the generator.

**Comment by lgirdwood** [comment]
@ShriramShastry any comments ?

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

**Comment by ShriramShastry** [comment]
> @ShriramShastry any comments ?

Tolerance checks with, impulse, and sine sample topologies can aid in general observation for both EQ FIR and IIR. performance criteria.

**Comment by singalsu** [comment]
> > @ShriramShastry any comments ?
> 
> Tolerance checks with impulse, and sine sample topologies can aid in general observation for both EQ FIR and IIR performance criteria. Example
> 

I discussed this offline with Sriram. There's no cmocka test for FIR and the existing test for IIR does only a chirp test. The test response is such that it amplifies some frequencies so it covers also saturation of PCM samples but extra coverage would not hurt. From above I  think the "Max output" test add would benefit both, especially FIR where the input sequence for max output is easy to generate.

But that's for other PR, since this PR is only about adding topology2 for existing EQ features.

**Comment by singalsu** [comment]
A new PR will replace this for IPC4 tplg2 blobs generation.

**Pull request closed without merging**